zipline/data/minute_bars.py

```diff
@@ -950,3 +957,107 @@ def load_raw_arrays(self, fields, start_dt, end_dt, sids):
                 out *= self._ohlc_inverse
             results.append(out)
         return results
+
+
+class PanelMinuteBarReader(object):
+    """
+    Reader for data passed as Panel.
+
+    DataPanel Structure
+    -------
+    items : Int64Index
+        Asset identifiers.  Must be unique.
+    major_axis : DatetimeIndex
+       Datetimes for data provided by the Panel.  Must be unique.
+    minor_axis : ['open', 'high', 'low', 'close', 'volume']
+       Price attributes.  Must be unique.
+
+    Attributes
+    ----------
+    The table with which this loader interacts contains the following
+    attributes:
+
+    panel : pd.Panel
+        The panel from which to read OHLCV data.
+    first_trading_day : pd.Timestamp
+        The first trading day in the dataset.
+    """
+    @preprocess(panel=call(verify_indices_all_unique))
+    def __init__(self, calendar, panel):
+
+        panel = panel.copy()
+        if 'volume' not in panel.minor_axis:
+            # Fake volume if it does not exist.
+            panel.loc[:, :, 'volume'] = int(1e9)
+
+        self.first_trading_day = pd.datetools.normalize_date(
+            panel.major_axis[0]
+        )
+        self._calendar = calendar
+
+        self.panel = panel
+
+        self._ohlc_inverse = 1. / OHLC_RATIO
+
+    @property
+    def last_available_dt(self):
+        return self.panel.major_axis[-1]
+
+    def load_raw_arrays(self, columns, start_dt, end_dt, assets):
```

Comment:
`PanelMinuteBarReader` alone has `self._ohlc_inverse`, but it can be removed if [this line](https://github.com/quantopian/zipline/blob/65a1e3465cf74f108be7ac9d881e265c277b390a/zipline/data/data_portal.py#L556) in `DataPortal` can be removed. That line assigns `self.MINUTE_PRICE_ADJUSTMENT_FACTOR`, which is [never used](https://github.com/quantopian/zipline/search?utf8=%E2%9C%93&q=MINUTE_PRICE_ADJUSTMENT_FACTOR&type=Code).

Everything else is roughly equivalent and can be consolidated. I suppose I'll put the `PanelBarReader` class in zipline/data/us_equity_pricing.py. Should I make it a subclass of `DailyBarReader`?